gobgpd/main.go

```diff
@@ -307,139 +283,170 @@ func main() {
-					},
-				}); err != nil {
-					log.Warn(err)
+				for _, dn := range newConfig.DynamicNeighbors {
+					log.Infof("Dynamic Neighbor %s is added to PeerGroup %s", dn.Config.Prefix, dn.Config.PeerGroup)
+					if _, err := apiServer.AddDynamicNeighbor(context.Background(), &api.AddDynamicNeighborRequest{
+						DynamicNeighbor: &api.DynamicNeighbor{
+							Prefix:    dn.Config.Prefix,
+							PeerGroup: dn.Config.PeerGroup,
+						},
+					}); err != nil {
+						log.Warn(err)
+					}
 				}
-			}
-			for _, p := range added {
-				log.Infof("Peer %v is added", p.State.NeighborAddress)
-				if _, err := apiServer.AddNeighbor(context.Background(), &api.AddNeighborRequest{
-					Peer: api.NewPeerFromConfigStruct(&p),
-				}); err != nil {
-					log.Warn(err)
+				for _, p := range added {
+					log.Infof("Peer %v is added", p.State.NeighborAddress)
+					if _, err := apiServer.AddNeighbor(context.Background(), &api.AddNeighborRequest{
+						Peer: api.NewPeerFromConfigStruct(&p),
+					}); err != nil {
+						log.Warn(err)
+					}
 				}
-			}
-			for _, p := range deleted {
-				log.Infof("Peer %v is deleted", p.State.NeighborAddress)
-				if _, err := apiServer.DeleteNeighbor(context.Background(), &api.DeleteNeighborRequest{
-					Peer: api.NewPeerFromConfigStruct(&p),
-				}); err != nil {
-					log.Warn(err)
+				for _, p := range deleted {
+					log.Infof("Peer %v is deleted", p.State.NeighborAddress)
+					if _, err := apiServer.DeleteNeighbor(context.Background(), &api.DeleteNeighborRequest{
+						Peer: api.NewPeerFromConfigStruct(&p),
+					}); err != nil {
+						log.Warn(err)
+					}
 				}
-			}
-			for _, p := range updated {
-				log.Infof("Peer %v is updated", p.State.NeighborAddress)
-				if u, err := apiServer.UpdateNeighbor(context.Background(), &api.UpdateNeighborRequest{
-					Peer: api.NewPeerFromConfigStruct(&p),
-				}); err != nil {
-					log.Warn(err)
-				} else {
-					updatePolicy = updatePolicy || u.NeedsSoftResetIn
+				for _, p := range updated {
+					log.Infof("Peer %v is updated", p.State.NeighborAddress)
+					if u, err := apiServer.UpdateNeighbor(context.Background(), &api.UpdateNeighborRequest{
+						Peer: api.NewPeerFromConfigStruct(&p),
+					}); err != nil {
+						log.Warn(err)
+					} else {
+						updatePolicy = updatePolicy || u.NeedsSoftResetIn
+					}
 				}
-			}
 
-			if updatePolicy {
-				if _, err := apiServer.SoftResetNeighbor(context.Background(), &api.SoftResetNeighborRequest{
-					Address:   "",
-					Direction: api.SoftResetNeighborRequest_IN,
-				}); err != nil {
-					log.Warn(err)
+				if updatePolicy {
+					if _, err := apiServer.SoftResetNeighbor(context.Background(), &api.SoftResetNeighborRequest{
+						Address:   "",
+						Direction: api.SoftResetNeighborRequest_IN,
+					}); err != nil {
+						log.Warn(err)
+					}
 				}
 			}
-		case <-sigCh:
-			apiServer.Shutdown(context.Background(), &api.ShutdownRequest{})
 		}
 	}
+
+	loop()
+
+	os.Exit(0)
```

Comment:
shouldn't need an Exit here if you modify the signal being captured to `os.Interrupt`